internal/pkg/service/stream/storage/statistics/repository/stats_allocation_test.go

```diff
@@ -0,0 +1,145 @@
+		require.NoError(t, err)
+		defer func() { require.NoError(t, session.Close()) }()
+		test.RegisterWriterVolumes(t, ctx, volumeRepo, session, 1)
+	}
+
+	// Create branch, source, sink, file, slice
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		branch := test.NewBranch(branchKey)
+		require.NoError(t, defRepo.Branch().Create(&branch, clk.Now(), by).Do(ctx).Err())
+		source := test.NewSource(sourceKey)
+		require.NoError(t, defRepo.Source().Create(&source, clk.Now(), by, "Create source").Do(ctx).Err())
+		sink := test.NewSinkWithLocalStorage(sinkKey)
+		require.NoError(t, defRepo.Sink().Create(&sink, clk.Now(), by, "Create sink").Do(ctx).Err())
+	}
+
+	// Create 2 more files/slices (3 totally)
+	// -----------------------------------------------------------------------------------------------------------------
+	var sliceKey1, sliceKey2 model.SliceKey
+	{
+		clk.Add(time.Hour)
+		require.NoError(t, fileRepo.Rotate(sinkKey, clk.Now()).Do(ctx).Err())
+
+		clk.Add(time.Hour)
+		require.NoError(t, fileRepo.Rotate(sinkKey, clk.Now()).Do(ctx).Err())
+
+		slices, err := sliceRepo.ListIn(sinkKey).Do(ctx).All()
+		require.NoError(t, err)
+		require.Len(t, slices, 3)
+		sliceKey1 = slices[0].SliceKey
+		sliceKey2 = slices[1].SliceKey
+	}
+
+	// Create stats records
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		assert.NoError(t, statsRepo.Put(ctx, []statistics.PerSlice{
+			{SliceKey: sliceKey1, Value: statistics.Value{
+				SlicesCount:      1,
+				FirstRecordAt:    utctime.MustParse("2000-01-01T01:00:00.000Z"),
+				LastRecordAt:     utctime.MustParse("2000-01-01T02:00:00.000Z"),
+				RecordsCount:     123,
+				UncompressedSize: 20000,
+				CompressedSize:   1000, // <<<<<<<<<<<
+			}},
+			{SliceKey: sliceKey2, Value: statistics.Value{
+				SlicesCount:      1,
+				FirstRecordAt:    utctime.MustParse("2000-01-01T02:00:00.000Z"),
+				LastRecordAt:     utctime.MustParse("2000-01-01T03:00:00.000Z"),
+				RecordsCount:     345,
+				UncompressedSize: 10000,
+				CompressedSize:   500,
+			}},
+		}))
+	}
+
+	// Move slice1 to storage.Staging, move slice2 to storage.Target
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		clk.Add(time.Hour)
+		require.NoError(t, sliceRepo.SwitchToUploading(sliceKey1, clk.Now()).Do(ctx).Err())
+		require.NoError(t, sliceRepo.SwitchToUploading(sliceKey2, clk.Now()).Do(ctx).Err())
+		require.NoError(t, sliceRepo.SwitchToUploaded(sliceKey1, clk.Now()).Do(ctx).Err())
+		require.NoError(t, sliceRepo.SwitchToUploaded(sliceKey2, clk.Now()).Do(ctx).Err())
+
+		clk.Add(time.Hour)
+		require.NoError(t, fileRepo.SwitchToImporting(sliceKey2.FileKey, clk.Now()).Do(ctx).Err())
+		require.NoError(t, fileRepo.SwitchToImported(sliceKey2.FileKey, clk.Now()).Do(ctx).Err())
```

Comment:
This refers to the fact that slice2 is in storage.Target? (Already on remote)

Reply:
Yes, I will add a comment.

Reply:
:thinking: Comment is already there 

```
// Move slice1 to storage.Staging, move slice2 to storage.Target
```